postRoutes.js

```diff
@@ -0,0 +1,103 @@
+const express = require("express");
+const router = express.Router();
+const db = require("./data/db.js");
+
+router.get("/", (request, response) => {
+  db.find()
+    .then(res => {
+      if (res) {
+        response.status(200).json({ success: true, res });
+      } else {
+        status(404).json({ success: false, message: "No posts available." });
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      response.status(500).json({ success: false, err });
+    });
+});
+
+router.get("/:id", (request, response) => {
+  const { id } = request.params;
+  db.findById(id)
+    .then(res => {
+      if (res) {
+        response.status(200).json({ success: true, res });
+      } else {
+        response
+          .status(404)
+          .json({ success: false, message: "Post does not exist." });
+      }
+    })
+    .catch(err => {
+      response.status(500).json({ success: false, err });
+    });
+});
+
+router.get("/:id/comments", (request, response) => {
+  const { id } = request.params;
+  db.findPostComments(id)
+    .then(res => {
+      response.status(200).json({ success: true, res });
+    })
+    .catch(err => {
+      response.status(500).json({ success: false, err });
+    });
+});
+
+router.post("/", (request, response) => {
+  const post = request.body;
+  db.insert(post)
+    .then(assignedId => {
+      response.status(201).json({ success: true, assignedId });
+    })
+    .catch(err => {
+      response.status(500).json({ success: false, err });
+    });
+});
+
+router.post("/:id/comments", (request, response) => {
+  //returns id of inserted comment
+  const post_id = request.params.id;
+  const { newComment } = request.body;
+
+  if (post_id) {
```

Comment:
This logic could be improved. post_id will always be received because it is coming from the url parameter. Meaning if there is no id, it will not hit this endpoint. Something you'd want to check is that the newComment had all of the required fields before submitting it to the database.